registration/include/pcl/registration/ppf_registration.h

```diff
@@ -56,15 +58,26 @@ namespace pcl
         */
       struct HashKeyStruct : public std::pair <int, std::pair <int, std::pair <int, int> > >
       {
+        HashKeyStruct () = default;
+
         HashKeyStruct(int a, int b, int c, int d)
         {
           this->first = a;
           this->second.first = b;
           this->second.second.first = c;
           this->second.second.second = d;
         }
+
+        std::size_t operator()(const HashKeyStruct& s) const noexcept
+        {
+            const std::size_t h1 = std::hash<int>{} (s.first);
+            const std::size_t h2 = std::hash<int>{} (s.second.first);
+            const std::size_t h3 = std::hash<int>{} (s.second.second.first);
+            const std::size_t h4 = std::hash<int>{} (s.second.second.second);
+            return h1 ^ (h2 << 1) ^ (h3 << 2) ^ (h4 << 3);
```

Comment:
Alternatively, we can cast `HashKeyStruct` to a `std::pair<int, std::pair<int, std::pair<int, int>>>` here and return the output of `boost::hash`.

Reply:
Yes, but to me looks good as it is now.